internal/services/containerapps/container_app_job_resource.go

```diff
@@ -93,15 +98,93 @@ func (r ContainerAppJobResource) Arguments() map[string]*schema.Schema {
 
 		"template": helpers.JobTemplateSchema(),
 
-		"secrets": helpers.SecretsSchema(),
+		"secret": helpers.SecretsSchema(),
+
+		"secrets": {
```

Comment:
This needs to be removed from the schema post 4.0, we usually do this by modifying the schema with the feature flag e.g. 

```suggestion
func (r ContainerAppJobResource) Arguments() map[string]*schema.Schema {
	schema := map[string]*pluginsdk.Schema{
			...
	}
	
	if !features.FourPointOhBeta() {
	schema["secrets"] = &pluginsdk.Schema{
			// secrets schema def here
		}
	}
	
	return schema
},
```